.github/workflows/image.yaml

```diff
@@ -0,0 +1,176 @@
+          - 525
+          - 535
+        flavor: 
+          - generic
+          - nvidia
+          - aws
+          - azure
+        ispr:
+          - ${{github.event_name == 'pull_request'}}
+        exclude:
+          - ispr: true
+            flavor: azure
+          - ispr: true
+            flavor: aws
+          - ispr: true
+            flavor: nvidia
+    steps:
+      - uses: actions/checkout@v4
+        name: Check out code
+      - name: Calculate build vars
+        id: vars
+        run: |
+          echo "COMMIT_SHORT_SHA=${GITHUB_SHA:0:8}" >> $GITHUB_ENV
+          echo "LOWERCASE_REPO_OWNER=$(echo "${GITHUB_REPOSITORY_OWNER}" | awk '{print tolower($0)}')" >> $GITHUB_ENV
+          REPO_FULL_NAME="${{ github.event.pull_request.head.repo.full_name }}"
+          echo "${REPO_FULL_NAME}"
+          echo "LABEL_IMAGE_SOURCE=https://github.com/${REPO_FULL_NAME}" >> $GITHUB_ENV
+
+          GENERATE_ARTIFACTS="false"
+          if [[ "${{ github.actor }}" == "dependabot[bot]" ]]; then
+            GENERATE_ARTIFACTS="false"
+          elif [[ "${{ github.event_name }}" == "pull_request" && "${{ github.event.pull_request.head.repo.full_name }}" == "${{ github.repository }}" ]]; then
+            GENERATE_ARTIFACTS="true"
+          elif [[ "${{ github.event_name }}" == "push" ]]; then
+            GENERATE_ARTIFACTS="true"
+          fi
+          echo "PUSH_ON_BUILD=${GENERATE_ARTIFACTS}" >> $GITHUB_ENV
+          echo "BUILD_MULTI_ARCH_IMAGES=${GENERATE_ARTIFACTS}" >> $GITHUB_ENV
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build base image and get kernel version
+        env:
+          IMAGE_NAME: ghcr.io/nvidia/driver
+          VERSION: ${COMMIT_SHORT_SHA}
+          BASE_TARGET: jammy
+        run: |
+          make DRIVER_BRANCH=${{ matrix.driver }} KERNEL_FLAVOR=${{ matrix.flavor }} build-base-jammy
+
+          trap "docker rm -f base-${BASE_TARGET}-${{ matrix.flavor }}" EXIT
+          docker run -d --name base-${BASE_TARGET}-${{ matrix.flavor }} ghcr.io/nvidia/driver:base-${BASE_TARGET}-${{ matrix.flavor }}-${{ matrix.driver }} 
+          # try 3 times every 10 seconds to get the file, if success exit the loop
+          for i in {1..3}; do
+              docker cp base-${BASE_TARGET}-${{ matrix.flavor }}:/var/kernel_version.txt kernel_version.txt && break
+              sleep 10
+          done
+      - name: Build image
+        env:
+          IMAGE_NAME: ghcr.io/nvidia/driver
+          VERSION: ${COMMIT_SHORT_SHA}
+          PRECOMPILED: "true"
+          DIST: signed_ubuntu22.04
+          CVE_UPDATES: "curl libc6"
+        run: |
+          source kernel_version.txt && \
```

Comment:
Does `kernel_version.txt` need to be "exported" from the previous step in some way?